### PR TITLE
Asset group sighting jobs

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Replaced the AssetGroupSightingJobs table with a json column in the AssetGroupSighting on the principle of implementing the simplest solution possible
- Replaced /api/v1/asset_group/{asset_group_guid}/{asset_group_sighting_guid}/commit with /api/v1/asset_group/sighting/{asset_group_sighting_guid}/commit as agreed
- Renamed "meta" in the AssetGroup and AssetGroupSighting with more meaningful  "config" 
---

### CODE REVIEW

File: app/modules/asset_groups/models.py [hunk 1]
```diff
@@ -846,11 +834,12 @@ def get_asset_for_file(self, filename):
     def begin_ia_pipeline(self, metadata):
         assert len(metadata.detection_configs) == 1
         assert metadata.data_processed == CreateAssetGroupMetadata.DataProcessed.complete
+        import copy
 
         for sighting_meta in metadata.request['sightings']:
             new_sighting = AssetGroupSighting(
                 asset_group_guid=self.guid,
-                meta=dict(sighting_meta),
+                config=copy.deepcopy(sighting_meta),
```
Comment (hwindsor) [inline app/modules/asset_groups/models.py:842]: Just doing dict() only copies the first level down, sighting_meta is deleted when the calling method goes out of scope so doing deepcopy is safer

File: app/modules/asset_groups/models.py [hunk 2]
```diff
@@ -128,12 +110,17 @@ class AssetGroupSighting(db.Model, HoustonModel):
     )
     asset_group = db.relationship('AssetGroup', backref=db.backref('sightings'))
 
-    meta = db.Column(db.JSON, nullable=True)
+    # configuration metadata from the create request
+    config = db.Column(db.JSON, nullable=True)
+
+    # May have multiple jobs outstanding, store as json array of job_ids
+    jobs = db.Column(db.JSON, nullable=True)
```
Comment (colinwkingen) [inline app/modules/asset_groups/models.py:117]: If we do make a broader Job or Task class for many houston initiated IA actions it would make sense to leverage it here. It would probably look similar to the AssetGroupJob, but would pack more variable information into a JSON field. That would be a separate PR of course and this simplification is good for now.

File: tests/modules/asset_groups/resources/test_commit_asset_group.py [hunk 3]
```diff
@@ -20,8 +20,13 @@ def test_commit_asset_group(flask_app_client, researcher_1, regular_user, test_r
         asset_group_uuid = response.json['guid']
         asset_group_sighting_guid = response.json['sightings'][0]['guid']
 
+        # Should not be able to commit as contributor
+        asset_group_utils.commit_asset_group_sighting(
+            flask_app_client, regular_user, asset_group_sighting_guid, 403
+        )
```
Comment (colinwkingen) [inline tests/modules/asset_groups/resources/test_commit_asset_group.py:26]: Theoretically the UI should prevent the user from seeing an option to do this, but it's good to have the check.

**Review by colinwkingen — APPROVED** [review]
LGTM

File: app/modules/asset_groups/models.py (hunk 2 above)
Comment (hwindsor) [reply app/modules/asset_groups/models.py:117]: Have a look at what I put for this in the IA Pipeline doc. I don't think there will need to be much more here.

File: tests/modules/asset_groups/resources/test_commit_asset_group.py (hunk 3 above)
Comment (hwindsor) [reply tests/modules/asset_groups/resources/test_commit_asset_group.py:26]: I work on the principle that while Bens UI is currently using this IF, If the IF is available for anyone to use, we need to be reasonably robust.
